components/scream/src/control/atmosphere_driver.cpp

```diff
@@ -323,11 +348,18 @@ void AtmosphereDriver::create_fields()
   // the internal fields/groups, and mark them as part of the RESTART group.
   for (const auto& f : m_atm_process_group->get_fields_in()) {
     const auto& fid = f.get_header().get_identifier();
-    auto fm = get_field_mgr(fid.get_grid_name());
+    const auto& gn = fid.get_grid_name();
+    if (fvphyshack and gn == "Physics CGLL") {
```

Comment:
I suspect this (as well as below) might have to be changed to "Physics GLL", to match what's used elsewhere.

Reply:
Good catch. That actually indicates that that workaround is not needed, assuming the restart file looks fine. I'll check and delete the workaround if it's fine.

Reply:
The restart file has no sign of the temporary CGLL fields, so the hack above (corrected) isn't needed. I don't actually understand why that's true, as I thought add_field'ing the temporary CGLL fields as Required would mean they'd appear in the restart file.

Reply:
That should be the case. Are the fields you refer to those that you later remove (with the newly added feature "remove_field" in AtmosphereProcess)?

Reply:
Oh, I see why. I had two layers of hacks, but only one was needed. The part of the code with the comment "// Don't save CGLL fields from ICs to the restart file." strips those fields from the restart file, so this first hack, whether written correctly or not, isn't needed.

Reply:
> That should be the case. Are the fields you refer to those that you later remove (with the newly added feature "remove_field" in AtmosphereProcess)?

That's not a great feature and probably something we should discuss that motivates a more flexible init process. In any case, that field removal I think would lead to a restart-file-write error later on if it weren't for the chunk of code starting with "// Don't save CGLL fields from ICs to the restart file", which makes the restart I/O object not see these fields.